internal/providers/cloudstackMetadata/cloudstack.go

```diff
@@ -0,0 +1,181 @@
+	mnt := ""
+
+	if !fileExists(configDriveMetadataMountPoint) {
+		err := error(nil)
+		mnt, err = mountConfigDrive("config")
+		if err != nil {
+			if mnt, err = mountConfigDrive("CONFIG"); err != nil {
+				return m, err
+			}
+		}
+		defer unmountConfigDrive(mnt)
+	} else {
+		mnt = configDriveMetadataMountPoint
+	}
+
+	if err := fetchAndSet("availability_zone.txt", mnt, "CLOUDSTACK_AVAILABILITY_ZONE", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	if err := fetchAndSet("instance_id.txt", mnt, "CLOUDSTACK_INSTANCE_ID", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	if err := fetchAndSet("service_offering.txt", mnt, "CLOUDSTACK_SERVICE_OFFERING", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	if err := fetchAndSet("cloud_identifier.txt", mnt, "CLOUDSTACK_CLOUD_IDENTIFIER", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	if err := fetchAndSet("local_hostname.txt", mnt, "CLOUDSTACK_LOCAL_HOSTNAME", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	if err := fetchAndSet("vm_id.txt", mnt, "CLOUDSTACK_VM_ID", m.Attributes); err != nil {
+		return providers.Metadata{}, err
+	}
+
+	keys, err := fetchKeys(mnt)
+	if err != nil {
+		return providers.Metadata{}, err
+	}
+	m.SshKeys = keys
+	return m, nil
+}
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	return (err == nil)
+}
+
+func labelExists(label string) bool {
+	_, err := getPath(label)
+	return (err == nil)
+}
+
+func getPath(label string) (string, error) {
+	path := diskByLabelPath + label
+
+	if fileExists(path) {
+		return path, nil
+	}
+
+	out, err := exec.Command("blkid", "-l", "-t", "LABEL="+label, "-o", "device").Output()
+	if err != nil {
+		return "", err
+	}
+
+	if len(out) > 0 {
+		return strings.TrimSpace(string(out)), nil
+	}
+
+	return "", fmt.Errorf("label not found: %s", label)
+}
+
+func mountConfigDrive(label string) (string, error) {
+	if !labelExists(label) {
+		return "", errors.New("Not able to find config drive.")
+	}
+	path, err := getPath(label)
+
+	mnt, err := ioutil.TempDir("", "coreos-metadata")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp directory: %v", err)
+	}
+
+	cmd := exec.Command("/bin/mount", "-o", "ro", "-t", "auto", path, mnt)
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+
+	return mnt, nil
+}
+
+func readFromConfigDrive(mnt string, file string) (string, bool, error) {
+	if !fileExists(filepath.Join(mnt, configDriveMetadataPath, file)) {
+		return "", false, nil
+	}
+
+	output, err := ioutil.ReadFile(filepath.Join(mnt, configDriveMetadataPath, file))
+
+	if err != nil {
+		return "", false, err
+	}
+
+	return string(output), true, nil
+}
+
+func unmountConfigDrive(mnt string) {
+	defer os.Remove(mnt)
+	defer syscall.Unmount(mnt, 0)
+}
+
+func fetchAndSet(key, mnt string, attrKey string, attributes map[string]string) error {
+	val, ok, err := readFromConfigDrive(mnt, key)
```

Comment:
Can we just use ReadFile directly here and check the result of [IsNotExist](https://golang.org/pkg/os/#IsNotExist)?